proposed/extended-coding-style-guide.md

```diff
@@ -122,22 +125,77 @@ declaration.
 
 There MUST be one `use` keyword per declaration.
 
-There MUST be one blank line after the `use` block.
+When using multiple classes, functions or constants within one namespace, you
+MUST group use statements within one namespace.
+
+Use statements MUST be in a block according to the entity (class, function or group)
+which is being grouped. Within each block there MUST be no blank lines. If a block
+has multiple lines there MUST be a blank line before the first line and a blank line
+after the last line.
+
+Classes, functions or constants grouped together into a single line must be listed
+alphabetically.
+
+The groups MUST be ordered such that classes are first, followed by functions and
+then constants.
 
 For example:
 
 ```php
 <?php
+declare(strict_types=1);
+
 namespace Vendor\Package;
 
-use FooClass;
-use BarClass as Bar;
-use OtherVendor\OtherPackage\BazClass;
+use Vendor\Package\{ClassA as A, ClassB, ClassC as C};
```

Comment:
Can we _explicitly_ disallow grouping namespaces. Eg instead of 

    use Symfony\Component\Console\{
        Input\InputInterface,
        Output\NullOutput,
        Output\OutputInterface,
    };

Where grouping is done on the level of the Console component, you MUST do

    use Symfony\Component\Console\Input\InputInterface;
    use Symfony\Component\Console\Output{
        NullOutput,
        OutputInterface,
    };

Called [compound namespaces](https://wiki.php.net/rfc/group_use_declarations#compound_namespaces) in the RFC.

The rationale behind this is that any commits adding or removing a use statement are likely to change the level at which the grouping is done, which leads to lots of unrelated changed lines.